openhtf/__init__.py

```diff
@@ -71,21 +72,33 @@ def PhaseTwo(test):
       # Analyze widget integration status
 
     Test(PhaseOne, PhaseTwo).Execute()
+
+  Note that Test() objects *must* be created in the main thread, but can be
+  .Execute()'d in a separate thread.
   """
 
   def __init__(self, *phases, **metadata):
     code_info = test_record.CodeInfo.ForModuleFromStack(levels_up=2)
+    self._test_options = TestOptions()
     self._test_info = TestData(phases, metadata=metadata, code_info=code_info)
-    self._output_callbacks = []
-    self.loop = False
+    self._lock = threading.Lock()
+    self._stopped = False
+    signal.signal(signal.SIGINT, self.Stop)
```

Comment:
I'm trying to steer away from global "framework" state and toward more cleanly encapsulated state.  Moving this to a top-level would force Executor to be a singleton.  There's still a ton of places where we force singleton use anyway, but I'd like to start migrating away from that model where it makes sense and is easy.

That said, to truly support multiple executors, this needs to be cleaned up a bit, I'll see what I can do.